src/coinbase/address/external_address.ts

```diff
@@ -0,0 +1,269 @@
+import { Amount, CoinbaseExternalAddressStakeOptions, StakeOptionsMode } from "../types";
+import { Coinbase } from "../coinbase";
+import Decimal from "decimal.js";
+import { Asset } from "../asset";
+import { StakingOperation } from "../staking_operation";
+
+/**
+ * A representation of a blockchain Address, which is a user-controlled account on a Network. Addresses are used to
+ * send and receive Assets. An ExternalAddress is an Address that is not controlled by the developer, but is instead
+ * controlled by the user.
+ */
+export class ExternalAddress extends Address {
+  /**
+   * Builds a stake operation for the supplied asset. The stake operation
+   * may take a few minutes to complete in the case when infrastructure is spun up.
+   *
+   * @param amount - The amount of the asset to stake.
+   * @param assetId - The asset to stake.
+   * @param options - Additional options for the stake operation.
+   * @returns {StakingOperation} The stake operation.
+   */
+  public async buildStakeOperation(
+    amount: Amount,
+    assetId: string,
+    options: CoinbaseExternalAddressStakeOptions = { mode: StakeOptionsMode.DEFAULT },
+  ): Promise<StakingOperation> {
+    await this.validateCanStake(amount, assetId, options);
+    return this.buildStakingOperation(amount, assetId, "stake", options);
+  }
+
+  /**
+   * Builds an unstake operation for the supplied asset.
+   *
+   * @param amount - The amount of the asset to unstake.
+   * @param assetId - The asset to unstake.
+   * @param options - Additional options for the unstake operation.
+   * @returns {StakingOperation} The unstake operation.
+   */
+  public async buildUnstakeOperation(
+    amount: Amount,
+    assetId: string,
+    options: CoinbaseExternalAddressStakeOptions = { mode: StakeOptionsMode.DEFAULT },
+  ): Promise<StakingOperation> {
+    await this.validateCanUnstake(amount, assetId, options);
+    return this.buildStakingOperation(amount, assetId, "unstake", options);
+  }
+
+  /**
+   * Builds a claim stake operation for the supplied asset.
+   *
+   * @param amount - The amount of the asset to claim stake.
+   * @param assetId - The asset to claim stake.
+   * @param options - Additional options for the claim stake operation.
+   * @returns {Transaction} The claim stake operation.
+   */
+  public async buildClaimStakeOperation(
+    amount: Amount,
+    assetId: string,
+    options: CoinbaseExternalAddressStakeOptions = { mode: StakeOptionsMode.DEFAULT },
+  ): Promise<StakingOperation> {
+    await this.validateCanClaimStake(amount, assetId, options);
+    return this.buildStakingOperation(amount, assetId, "claim_stake", options);
+  }
+
+  /**
+   * Get the stakeable balance for the supplied asset.
+   *
+   * @param asset_id - The asset to check the stakeable balance for.
+   * @param options - Additional options for getting the stakeable balance.
+   * @returns {string} The stakeable balance.
+   */
+  public async getStakeableBalance(
```

Comment:
this should be stakeableBalance https://docs.google.com/document/d/1Z06zRTxFREvOG4L4dYIfZW6ItW1I7tiYupshaf4KsYU/edit

fyi @erdimaden

Reply:
Should the other balances also then have the `get` removed from the name?

Reply:
Renamed, please take a look.